innsending/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/innsending/RelativlyGenericInnsendingProcessor.kt

```diff
@@ -0,0 +1,149 @@
+        val uuid: String = message[Key.UUID.str].asText()
+        when (transaction) {
+            Transaction.NEW -> {
+                rapidsConnection.publish(
+                    JsonMessage.newMessage(
+                        mapOf(
+                            Key.EVENT_NAME.str to event.name,
+                            Key.BEHOV.str to listOf(BehovType.VIRKSOMHET.name),
+                            Key.ORGNRUNDERENHET.str to message[Key.ORGNRUNDERENHET.str].asText(),
+                            Key.UUID.str to uuid
+                        )
+                    ).toJson()
+                )
+                rapidsConnection.publish(
+                    JsonMessage.newMessage(
+                        mapOf(
+                            Key.EVENT_NAME.str to event.name,
+                            Key.BEHOV.str to listOf(BehovType.ARBEIDSFORHOLD.name),
+                            Key.IDENTITETSNUMMER.str to message[Key.IDENTITETSNUMMER.str].asText(),
+                            Key.UUID.str to uuid
+                        )
+                    ).toJson()
+                )
+            }
+            Transaction.IN_PROGRESS -> {
+                if (isDataCollected(*step1data(message[Key.UUID.str].asText()))) {
+                    rapidsConnection.publish(
+                        JsonMessage.newMessage(
+                            mapOf(
+                                Key.EVENT_NAME.str to event.name,
+                                Key.BEHOV.str to listOf(BehovType.PERSISTER_IM.name),
+                                Key.INNTEKTSMELDING.str to customObjectMapper().readTree(redisStore.get(uuid + DataFelter.INNTEKTSMELDING_REQUEST.str)!!),
+                                Key.UUID.str to uuid
+                            )
+                        ).toJson()
+                    )
+                }
+            }
+            Transaction.FINALIZE -> {
+                println("I was not supposed to be hereeeeeeeeeeeeeeeeeeeeeee")
+            }
+            Transaction.TERMINATE -> {}
+        }
+    }
+
+    fun finalize(message: JsonMessage) {
+        redisStore.set(message[Key.UUID.str].asText(), message[Key.INNTEKTSMELDING_DOKUMENT.str].asText())
+        rapidsConnection.publish(
+            JsonMessage.newMessage(
+                mapOf(
+                    Key.EVENT_NAME.str to EventName.INNTEKTSMELDING_MOTTATT,
+                    Key.INNTEKTSMELDING_DOKUMENT.str to message[Key.INNTEKTSMELDING_DOKUMENT.str].asText(),
+                    Key.UUID.str to message[Key.UUID.str]
+                )
+            ).toJson()
+        )
+    }
+
+    fun startStransactionIfAbsent(message: JsonMessage): Transaction {
+        if (isFailMelding(message)) {
+            return Transaction.TERMINATE
+        }
+        val uuid = message.get(Key.UUID.str).asText()
+        val eventKey = "${uuid}${event.name}"
+        val value = redisStore.get(eventKey)
+        println("uuuiiidd  " + value + "EventKey " + eventKey)
```

Comment:
litt logging som kan ryddes opp i, men dette er vel mer Proof-Of-Concept foreløpig uansett